usermods/M5Stick2_Visualiser/M5Stick2_Visualiser.h

```diff
@@ -0,0 +1,312 @@
+        PinManagerPinType displayPins[] = { { TFT_CS, true}, { TFT_DC, true}, { TFT_RST, true }, { TFT_BL, true } };
+        DEBUG_PRINTLN("TFT try to allocate display");
+        if (!pinManager.allocateMultiplePins(displayPins, sizeof(displayPins)/sizeof(PinManagerPinType), PinOwner::UM_FourLineDisplay)) {
+            pinManager.deallocateMultiplePins(spiPins, 3, PinOwner::HW_SPI);
+            enabled = false;
+            return;
+        }
+        DEBUG_PRINTLN("TFT init");
+
+        tft.init();
+        tft.setRotation(0);  //Rotation here is set up for the text to be readable with the port on the left. Use 1 to flip.
+        tft.fillScreen(TFT_BLACK);
+
+        if (TFT_BL >= 0) 
+        {
+            pinMode(TFT_BL, OUTPUT); // Set backlight pin to output mode
+            digitalWrite(TFT_BL, HIGH); // Turn backlight on.
+        }
+    }
+
+    /*
+     * connected() is called every time the WiFi is (re)connected
+     * Use it to initialize network interfaces
+     */
+    void connected() {
+      //Serial.println("Connected to WiFi!");
+    }
+
+    /*
+     * loop() is called continuously. Here you can check for events, read sensors, etc.
+     *
+     * Tips:
+     * 1. You can use "if (WLED_CONNECTED)" to check for a successful network connection.
+     *    Additionally, "if (WLED_MQTT_CONNECTED)" is available to check for a connection to an MQTT broker.
+     *
+     * 2. Try to avoid using the delay() function. NEVER use delays longer than 10 milliseconds.
+     *    Instead, use a timer check as shown here.
+     */
+    void loop() {
+        char buff[LINE_BUFFER_SIZE];
+
+        tft.setTextSize(2);
+
+        // Wifi name
+        tft.setTextColor(TFT_GREEN);
+        tft.setCursor(0, 60);
+
+        // Check if we time interval for redrawing passes.
+        if (millis() - lastUpdate < USER_LOOP_REFRESH_RATE_MS)
+        {
+            return;
+        }
+        //tft.fillScreen(TFT_BLACK);
+        lastUpdate = millis();
+  
+        // Turn off display after 5 minutes with no change.
+        if (!displayTurnedOff && millis() - lastRedraw > 5*60*1000)
+        {
+            DEBUG_PRINTLN("Turn off display");
+            if (TFT_BL >= 0) digitalWrite(TFT_BL, LOW); // Turn backlight off. 
+            displayTurnedOff = true;
+        } 
+
+        lastRedraw = millis();
+
+        knownBrightness = bri;
+        knownMode = strip.getMainSegment().mode;
+        knownPalette = strip.getMainSegment().palette;
+        knownEffectSpeed = strip.getMainSegment().speed;
+        knownEffectIntensity = strip.getMainSegment().intensity;
+        tft.setTextSize(1);
+        tft.setTextColor(TFT_GREEN);
+        tft.setTextFont(1);
+        tft.setCursor(100, 100);
+
+        for(int i = 0; i < 16; i++) {
+            tft.fillRect(fftResult[i] / 2, i*15 + 3, 135 - (fftResult[i] / 2), 10, TFT_BLACK);
+            tft.fillRect(0, i*15 + 3, fftResult[i] / 2, 10, TFT_GOLD);
```

Comment:
This way of referencing a variable inside the audioreactive usermod may not work in the future, when each usermod will have it's own namespace. The safer way is to use a local pointer

```c++
  um_data_t *um_data = getAudioData();
  uint8_t *fftChannels = (uint8_t*)um_data->u_data[2];
```

Then you can simply use `fftChannels[i]` to access the data. If audioreactive was disabled, you will automatically get "soundsim" like in any other effect.